src/Microsoft.AspNet.Mvc.Razor/Razor/PreCompileViews/PrecompilationCacheEntry.cs

```diff
@@ -0,0 +1,72 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using Microsoft.CodeAnalysis;
+
+namespace Microsoft.AspNet.Mvc.Razor
+{
+    /// <summary>
+    /// Represents an entry in the cache used by <see cref="RazorPreCompiler"/>.
+    /// </summary>
+    public class PrecompilationCacheEntry
+    {
+        /// <summary>
+        /// Initializes a new instance of <see cref="PrecompilationCacheEntry"/> that represents a successful
+        /// parse.
+        /// </summary>
+        /// <param name="diagnostics">The <see cref="IReadOnlyList{Diagnostic}"/>s produced from parsing the Razor
+        /// file.</param>
+        public PrecompilationCacheEntry([NotNull] RazorFileInfo fileInfo,
+                                             [NotNull] SyntaxTree syntaxTree)
+        {
+            FileInfo = fileInfo;
+            SyntaxTree = syntaxTree;
+        }
+
+        /// <summary>
+        /// Initializes a new instance of <see cref="PrecompilationCacheEntry"/> that represents a failure
+        /// in Razor parsing.
```

Comment:
simplify wording and make consistent w/ other overload: "that represents a failure in Razor parsing" -> "for a failed parse"